fn_phish_tank/fn_phish_tank/components/fn_phish_tank_submit_url.py

```diff
@@ -0,0 +1,92 @@
+
+
+class FunctionComponent(ResilientComponent):
+    """Component that implements Resilient function 'fn_phish_tank_submit_url"""
+
+    def __init__(self, opts):
+        """constructor provides access to the configuration options"""
+        super(FunctionComponent, self).__init__(opts)
+        self.options = opts.get("fn_phish_tank", {})
+        selftest.selftest_function(opts)
+
+    @handler("reload")
+    def _reload(self, event, opts):
+        """Configuration options have changed, save new values"""
+        self.options = opts.get("fn_phish_tank", {})
+
+    @function("fn_phish_tank_submit_url")
+    def _fn_phish_tank_submit_url_function(self, event, *args, **kwargs):
+        """Function: Function to check URLs against PhishTanks(https://www.phishtank.com/) Database to see URL is flagged as phishing or not phishing."""
+        try:
+            # Get the function parameters:
+            phish_tank_check_url = kwargs.get("phish_tank_check_url")  # text
+
+            # Get function parameters from config file.
+            phish_tank_api_url = self.options.get('phish_tank_api_url')
+            phish_tank_api_key = self.options.get('phish_tank_api_key')
+            proxy = self.options.get('proxy')
+
+            log = logging.getLogger(__name__)
+            log.info("PhishTank Check URL: %s", phish_tank_check_url)
+            log.info("PhishTank API Access URL: %s", phish_tank_api_url)
+            log.info("PhishTank API Access KEY: %s", phish_tank_api_key)
+            log.info("Proxy Server Address: %s", proxy)
+
+            yield StatusMessage("Checking Phishing Status Against URL: {}".format(phish_tank_check_url))
+
+            # Initialing the resilient result object
+            _result_obj = ResultPayload('fn_phish_tank',
+                                        function_inputs=[phish_tank_api_url, phish_tank_check_url, phish_tank_api_key])
+
+            # PhihshTank Proxy Data
+            pt_proxy = phish_tank_helper.format_proxy_data(proxy_data=proxy)
+
+            # PhishTank POST Call Data
+            pt_post_call_data = phish_tank_helper.create_post_data(phish_tank_check_url, api_key=phish_tank_api_key)
+
+            # Initialising the Request Session
+            pt_helper_class_instance = phish_tank_helper()
+            _request_session = pt_helper_class_instance.session()
+
+            # Making Post request to PhishTank Database for check Phishing status against given URL
+            _api_response = _request_session.post(phish_tank_api_url, data=pt_post_call_data, proxies=proxy)
+
+            _api_response.raise_for_status()
+
+            # Converting received response to json format
+            _api_response_json = _api_response.json()
+
+            # Converting string time to epoch format
+            _received_time_stamp = _api_response_json.get('meta').get('timestamp')
+            _epoch_time = phish_tank_helper.timestamp_to_ms_epoch(_received_time_stamp)
```

Comment:
Instead of creating a new variable `_received_time_stamp ` I think you could just do 

`_epoch_time = phish_tank_helper.timestamp_to_ms_epoch(_api_response_json.get('meta').get('timestamp'))`

Reply:
I was doing in the same way, marks suggested not to modified original value. needs to add additional param in the last comment.
see-https://github.com/ibmresilient/resilient-community-apps/pull/37#discussion_r277299729

Reply:
Hi Neetin,
I reviewed the linked comment and believe they are for a different issue. 
I will explain. In the comment you linked you are doing `_api_response_json['meta']['timestamp'] = _epoch_time` which replaces `_api_response_json['meta']['timestamp']` with the value of `_epoch_time`. 
In that linked comment mark wanted you to instead add a new attribute to `_api_response_json`. He wanted you to do `_api_response_json['meta']['timestamp_epoch'] = _epoch_time` instead so that the original value of `_api_response_json['meta']['timestamp'] ` is not lost.

This would be different. Here you are creating a variable which copies a value from `_api_response_json` and then defining another variable. In this case it may make more sense to just do 
```
_epoch_time = phish_tank_helper.timestamp_to_ms_epoch(_api_response_json.get('meta').get('timestamp'))
```
because this way you have 1 less variable defined and the original input remains unchanged.

Reply:
ok